ci/ray_ci/linux_container.py

```diff
@@ -0,0 +1,95 @@
+import os
+import subprocess
+import sys
+from typing import List, Optional
+
+from ci.ray_ci.container import Container
+
+_DOCKER_ENV = [
+    "BUILDKITE_BUILD_URL",
+    "BUILDKITE_BRANCH",
+    "BUILDKITE_COMMIT",
+    "BUILDKITE_JOB_ID",
+    "BUILDKITE_LABEL",
+    "BUILDKITE_PIPELINE_ID",
+]
+_DOCKER_CAP_ADD = [
+    "SYS_PTRACE",
+    "SYS_ADMIN",
+    "NET_ADMIN",
+]
+
+
+class LinuxContainer(Container):
+    def __init__(
+        self,
+        docker_tag: str,
+        volumes: Optional[List[str]] = None,
+        envs: Optional[List[str]] = None,
+    ) -> None:
+        super().__init__(docker_tag)
+        self.volumes = volumes or []
+        self.envs = envs or []
+        self.envs += _DOCKER_ENV
+
+    def install_ray(self, build_type: Optional[str] = None) -> List[str]:
+        env = os.environ.copy()
+        env["DOCKER_BUILDKIT"] = "1"
+        subprocess.check_call(
+            [
+                "docker",
+                "build",
+                "--pull",
+                "--build-arg",
+                f"BASE_IMAGE={self._get_docker_image()}",
+                "--build-arg",
+                f"BUILD_TYPE={build_type or ''}",
+                "-t",
+                self._get_docker_image(),
+                "-f",
+                "/ray/ci/ray_ci/tests.env.Dockerfile",
+                "/ray",
+            ],
+            env=env,
+            stdout=sys.stdout,
+            stderr=sys.stderr,
+        )
+
+    def _get_run_command(
+        self,
+        script: List[str],
```

Comment:
when sending in, should script be a string here? the `"\n".join(script)` is duplicated now, and "script" for running in the env, is actually a text?

the reason that you have a list of strings, is only the ease of the external caller.

I feel that you should keep the `_get_run_command`, and have another function called `wrap_

Reply:
let me break this function smaller and reuse the '\n'.join(script) part. It passes down as an array as it comes from a couple of public API that has the script in the array format.